pandas/_libs/interval.pyx

```diff
@@ -91,8 +91,18 @@ cdef class Interval(IntervalMixin):
         # constructor (__cinit__/__new__) to avoid them
         if closed not in _VALID_CLOSED:
             raise ValueError("invalid option for 'closed': %s" % closed)
-        if not left <= right:
-            raise ValueError('left side of interval must be <= right side')
+        try:
+            if not left <= right:
+                raise ValueError('left side of interval must be <= right side')
+        except TypeError as e:
+            msg = 'left and right must be comparable, {e}'.format(e=e)
+            raise TypeError(msg)
+        if isinstance(left, Timestamp) and left.tzinfo != right.tzinfo:
+            # GH 18538
+            msg = ("left and right must have the same time zone, got "
+                   "'{left_tz}' and '{right_tz}'").format(
+                   left_tz=left.tzinfo, right_tz=right.tzinfo)
+            raise ValueError(msg)
```

Comment:
What about pytz tzinfos? Eg US/Eastern is represented by a different tzinfo object for DST and non-DST

Reply:
Does my switch to using `get_timezone` fix this?  The docstring seems to indicate so, but I'm not super familiar with this part of the codebase.

Reply:
I think that should do it, thanks.